### DISCUSSION TITLE
Running the combined image locally

### DISCUSSION THREAD

**Discussion by gaow** [statgenetics/statgen-courses#56] [General]
@yiweizh-memverge to fulfill the request to support running it locally, I have some codes here: 

```bash
set -e
# Get the relevant image
docker pull yiweizh/rockefeller-jupyter:latest
# Stop currently running servers to start from scratch if they are the same name; but don't stop any running instances (line below commented out)
docker container stop $(docker container ls -q --filter name=statgen_course_gw) &> /dev/null || true
# Create a user folder for notebook instance and set the permission
mkdir -p $HOME/statgen_course_gw && (chmod g=u $HOME/statgen_course_gw || true)
docker run --rm -it  -p 8888:8888 -v /tmp:/tmp -v $HOME/statgen_course_gw:/home/jovyan/handson-tutorial --name statgen_course_gw  -e NB_UID=1000 yiweizh/rockefeller-jupyter
```

Then follow the prompts to open the URL. However it does not work -- I get stuck at login page. 

Does it work on your end? Could you please take a look and see if you could fix it? Thanks!

**Comment by danielnachun** [comment]
I think this needs `-p 8888:8888` to work correctly locally.  That argument forwards the port 8888 into the container.  When I add that the above command seems to work as expected.

**Reply by gaow** [reply]
thanks @danielnachun for testing! Hmm I previously when I tried 8888 it's already being used on my computer so did something like `-p <port>:8888` where `port` is randomly generated, thinking that should be fine. However, that did not help. But maybe I need to have a cleaner starting point on my computer to test it. Too much going on with my system earlier. Will try that tmr.

**Reply by gaow** [reply]
Updated the command above anywas with `-p 8888:8888` -- @dianacornejo please test

**Reply by yiweizh-memverge** [reply]
Thanks @danielnachun for answering! Yes, `-p 8888:8888` would be required.

**Reply by gaow** [reply]
I confirm that `-p 8888:8888` works. However port 8888 may not always be available. THat was why i tried `-p <random_port>:8888`.

I don't think this would be a so frequently used feature considering we have the cloud setup. So if people run into an issue with port being otherwise engaged we can always provide tips to close ports to free it up, @yiweizh-memverge just in case some students try to do this and ask you the question.
